share/spice/hackage/hackage.js

```diff
@@ -0,0 +1,102 @@
+// curl https://www.haskell.org/hoogle/\?mode\=json\&hoogle\=json
+// API response format.
+// {"results" => [
+//    {"location"=>"http://hackage.haskell.org/package/JSON-Combinator",
+//     "self"=>"package JSON-Combinator",
+//     "docs"=> "A combinator library on top of a generalised JSON type.\n\nVersion 0.2.8 "}]}
+
+(function (env) {
+    "use strict";
+    env.ddg_spice_hackage = function(api_result) {
+
+        // get documentation string
+        function docString(doc) {
```

Comment:
Let's define these private functions outside the callback, above `env.ddg_spice_hackge`. We're working in a closure so they'll be accessible in the `ddg_spice_hackge` but won't get added to the global `env` :smiley: